foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/sessions/UnitOfWorkImpl.java

```diff
@@ -5579,6 +5644,7 @@ public void iterate(Object object) {
                 // PERF: Avoid initialization of new objects if none.
                 if (hasNewObjects()) {
                     Object original = getNewObjectsCloneToOriginal().remove(object);
+                    removeObjectFromPrimaryKeyToNewObjects(object, primaryKey);
```

Comment:
What about `protected void preMergeChanges() {` line 3870...3886 maybe `removeObjectFromPrimaryKeyToNewObjects(...)` should be called too.

Reply:
You are right. I've added a call to `removeObjectFromPrimaryKeyToNewObjects`